Src/main.c

```diff
@@ -103,13 +103,28 @@ int main(void)
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
+  //Initialize ADC and TIM
+  HAL_ADC_Init(&hadc);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  //set servo PWM output relative to input voltage from potentiometer.
+  //potentiometer voltage from 0-3.3V, PWM modulation from 1-2ms on-time at 50Hz
+  double ADCconvert = 1023; //converting ADC to decimal
+  double PeriodToOnTime = 3000; //3000 to 1ms
   while (1)
   {
-  /* USER CODE END WHILE */
+      //Updating timer parameters:
+      //ADC conversion:
+      double OnTime = HAL_ADC_GetValue(&hadc); //RAW ADC input
+      OnTime = OnTime / ADCconvert; //Converting raw to decimal of input range
+      OnTime++; //incrementing as minimum on time is 1ms
```

Comment:
Make sure you set these as consts. Ideally, the consts you should be using are:
- SERVO_PWM_MIN
- SERVO_PWM_MAX
- ADC_MAX_VALUE (already in yours)

Whether these values equal a percentage of the duty cycle or are in milliseconds is up to you, but think of it this way: If I asked you to use a servo with a 1.5ms to 2.5ms control range, how many lines would your code have to change? The answer should (almost always) be very little.

Reply:
Added the max function, now would need to change 2 numbers.